examples/hello-world/README

```diff
@@ -0,0 +1,18 @@
+First, build the DBSP docker container following the instructions
```

Comment:
`examples` is a special folder where cargo looks for when one runs `cargo run --example`.  We shouldn't abuse it for end-to-end examples.  The right thing to do is to reorganize the project moving all crates to a separate `crates` folder so we can have end-to-end `examples` here.  We can do this once other outstanding PRs have been merged. Maybe we can keep it with other projects in `pipeline_manager/demo` for now?